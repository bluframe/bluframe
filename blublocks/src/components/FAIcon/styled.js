// @flow

import styled, { type StyledComponent } from "styled-components"
import type { Theme } from "@bluframe/blublocks"

export const Icon: StyledComponent<
  { +fontSize: string },
  Theme,
  HTMLElement
> = styled.i`
  color: ${({ theme }) => theme.palette.primary.main};
  font-size: ${({ fontSize }) => fontSize};
`
