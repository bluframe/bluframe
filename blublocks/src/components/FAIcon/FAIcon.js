// @flow

import { type ComponentProps, DEFAULT_FONT_SIZE } from "."
import React from "react"
import { Icon as StyledIcon } from "./styled"

const BluFrameFAIcon = ({
  name,
  fontSize = DEFAULT_FONT_SIZE
}: ComponentProps): React$Node => (
  <StyledIcon className={`fa fa-${name} fa-2-5x`} fontSize={fontSize} />
)

export default BluFrameFAIcon
