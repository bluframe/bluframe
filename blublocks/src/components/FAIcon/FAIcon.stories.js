// @flow

import FAIcon, { type ComponentProps } from "components/FAIcon"
import React from "react"

const stories = {
  component: FAIcon,
  title: "Components/FAIcon"
}

const Template = (args: ComponentProps) => <FAIcon {...args} />

export const Default: any = Template.bind({})
Default.args = {
  name: "angle-double-up"
}

export const Big: any = Template.bind({})
Big.args = {
  fontSize: "40px",
  name: "angle-double-up"
}

export default stories
