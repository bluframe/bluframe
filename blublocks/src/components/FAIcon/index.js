// @flow

import "./font-awesome.min.css"
import "./font-awesome.css"
import FAIcon from "./FAIcon"

export type ComponentProps = {
  name: string,
  fontSize?: string
}

export const DEFAULT_FONT_SIZE = "20px"

export default FAIcon
