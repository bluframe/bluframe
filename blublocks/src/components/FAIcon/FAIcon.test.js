// @flow

import FAIcon from "./FAIcon"
import React from "react"
import { render } from "tests"

describe("BluFrame", () => {
  describe("FAIcon", () => {
    it("renders default size", () => {
      const props = {
        name: "angle-double-up"
      }

      const { container } = render(<FAIcon {...props} />)

      expect(container.firstChild).toMatchSnapshot()
    })

    it("renders custom size", () => {
      const props = {
        fontSize: "30px",
        name: "angle-double-up"
      }

      const { container } = render(<FAIcon {...props} />)

      expect(container.firstChild).toMatchSnapshot()
    })
  })
})
