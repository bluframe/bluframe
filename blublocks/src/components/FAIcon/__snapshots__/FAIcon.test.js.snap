// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BluFrame FAIcon renders custom size 1`] = `
<i
  class="styled__Icon-sc-ayi7v7-0 diTGXO fa fa-angle-double-up fa-2-5x"
  font-size="30px"
/>
`;

exports[`BluFrame FAIcon renders default size 1`] = `
<i
  class="styled__Icon-sc-ayi7v7-0 diTHrz fa fa-angle-double-up fa-2-5x"
  font-size="20px"
/>
`;
